### PR TITLE
Initial creation of the Water Spring Config Server

### PR DESCRIPTION
**Pull request by zmoore-usgs**

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+server:
+  port: ${serverPort:8888}
+  use-forward-headers: true
+  ssl:
+    key-store: ${keystoreLocation}
+    key-store-password: ${keystorePassword}
+    key-password: ${keystorePassword}
+    key-alias: ${keystoreSSLKey:tomcat}
```
Comment (isuftin) [inline src/main/resources/application.yml]: Is there anything to do with Tomcat here?

Comment (zmoore-usgs) [reply src/main/resources/application.yml]: The Spring Boot services we write generally use tomcat internally for their embedded web server and that is the default SSL key alias it looks for unless you override it in configuration.

**Comment by isuftin** [comment]
@zmoore-usgs That is, in fact, what caused the issue.

This seems to work...

```
mvn clean package -DoutputDirectory=.
version=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec)
cp "./target/water-spring-config-server-${version}.jar" ./water-spring-config-server.jar
openssl genrsa -out .travis/wildcard.key 2048
openssl req -nodes -newkey rsa:2048 -keyout  .travis/wildcard.key -out  .travis/wildcard.csr -subj "/C=US/ST=Wisconsin/L=Middleon/O=US Geological Survey/OU=WMA/CN=*"
openssl x509 -req -days 9999 -in .travis/wildcard.csr -signkey .travis/wildcard.key  -out .travis/wildcard.crt
openssl pkcs12 -export -in ./.travis/wildcard.crt -inkey ./.travis/wildcard.key -name water-spring-config-server -out ./.travis/wildcard.p12 -passout pass:changeit
keytool -importkeystore -deststorepass changeit -destkeystore ./.travis/cacerts -srckeystore ./.travis/wildcard.p12 --srcstorepass "changeit" -destalias tomcat
cp ${JAVA_HOME}/jre/lib/security/cacerts ./.travis
mkdir -p ./.travis/secrets/test-app/
mkdir -p ./.travis/configs/test-app/
java -jar -Djava.security.egd=file:/dev/./urandom ./water-spring-config-server.jar \
--gitSecretRepoUri="file://./.travis/secrets/test-app/" \
--gitConfigRepoUri="file://./.travis/configs/test-app/" \
--gitSecretRepoUser="secrets_user" \
--gitSecretRepoPassword="secrets_user_password" \
--gitConfigRepoUser="config_user" \
--gitConfigRepoPassword="config_password" \
--configServerUser="config_server_user" \
--configServerPassword="config_server_password" \
--keystoreLocation="./.travis/cacerts" \
--keystorePassword="changeit" \
--keystoreSSLKey="water-spring-config-server"

```

**Comment by isuftin** [comment]
@zmoore-usgs Does this application have a health check of any kind? All I could find is..

```
curl -u config_server_user:config_server_password -k https://localhost:8888/actuator/health


{"status":"DOWN"}
```

**Comment by zmoore-usgs** [comment]
@isuftin Yeah, the way we've done other Spring services in the past has been to assume the keystore and SSL key have the same password, so I continued that here.

In terms of a health check it sounds like there's currently a bug preventing that from working correctly..

https://github.com/spring-cloud/spring-cloud-config/issues/269

Might be worth implementing some sort of health check on our own

**Comment by zmoore-usgs** [comment]
Sounds like it does have this though: https://cloud.spring.io/spring-cloud-config/single/spring-cloud-config.html#_health_indicator

**Comment by zmoore-usgs** [comment]
I haven't tried configuring that

**Comment by isuftin** [comment]
@zmoore-usgs I've added a PR to your fork that includes the Travis config for this. The Travis config  builds, configures the server and launches it in the background. It tests the output from the server to verify it's running then performs a healthcheck.

https://travis-ci.org/USGS-CIDA/water-spring-config-server/builds/401961296?utm_source=github_status&utm_medium=notification

This PR looks good if you can merge my PR to your branch before merging this one. 

I think going forward we will want to deal with the password issue for the SSL key being imported. I don't know that we can or want to guarantee that the incoming key has a password and it matches what happens to be the java keystore password.

This Travis config also does not test getting configs from the server. The application.yml files are blank. This is a minimum setup script to get a server running which, for the initial commit, seems to be what you want.

**Comment by zmoore-usgs** [comment]
Closing in favor of #2

**Pull request closed without merging**